sqlglot/dialects/bigquery.py

```diff
@@ -293,6 +293,14 @@ class BigQuery(Dialect):
     # All set operations require either a DISTINCT or ALL specifier
     SET_OP_DISTINCT_BY_DEFAULT = dict.fromkeys((exp.Except, exp.Intersect, exp.Union), None)
 
+    ANNOTATORS = {
+        **Dialect.ANNOTATORS,
+        exp.Abs: lambda self, e: self._annotate_by_args(e, "this"),
+        exp.Sign: lambda self, e: self._annotate_by_args(e, "this"),
+        exp.Greatest: lambda self, e: self._annotate_by_args(e, "this", "expressions"),
+        exp.Least: lambda self, e: self._annotate_by_args(e, "this", "expressions"),
```

Comment:
Are these actually BigQuery-specific? My intuition is that these should be true for most dialects.

Reply:
In general yeah, functions like ABS, SIGN, FLOOR etc should be input dependent for most dialects so it could be a safe bet. Should we move these to `Dialect` now or extrapolate the common patterns once we have sufficient coverage for a few dialects?

Reply:
I'd check a few "important" dialects to make sure this is true for them and then move these annotators out.